pandas/tests/frame/test_analytics.py

```diff
@@ -1805,6 +1805,21 @@ def test_built_in_round(self):
             {'col1': [1., 2., 3.], 'col2': [1., 2., 3.]})
         tm.assert_frame_equal(round(df), expected_rounded)
 
+    def test_round_nonunique_categorical(self):
+        # See GH21809
+        idx = pd.CategoricalIndex(['low'] * 3 + ['hi'] * 3)
+        df = pd.DataFrame(np.random.rand(6, 3), columns=list('abc'))
+
+        truth = df.round(3)
+        truth.index = idx
+
+        dfb = df.copy().set_index(idx)
+        assert dfb.shape == (6, 3)
+        dfb_round = dfb.round(3)
+        assert dfb_round.shape == (6, 3)
+
+        tm.assert_frame_equal(truth, dfb_round)
```

Comment:
can you use the standard format we have, e.g.
```
result = 
expected = 
tm.assert_frame_equal(result, expected)
```

Reply:
Done